src/java/picard/illumina/ExtractIlluminaBarcodes.java

```diff
@@ -73,19 +73,52 @@
-                "    * matched barcode sequence\n" +
-                "Note 1: that the order of specification of barcodes can cause arbitrary differences in output for poorly matching barcodes.\n" +
-                "Note 2: molecular barcodes (M in the read structure) are not the barcode being extracted here and will be ignored here.\n\n",
-        usageShort = "Tool to determine the barcode for each read in an Illumina lane",
+
+        usage = ExtractIlluminaBarcodes.USAGE_SUMMARY + ExtractIlluminaBarcodes.USAGE_DETAILS,
+        usageShort = ExtractIlluminaBarcodes.USAGE_SUMMARY,
         programGroup = Illumina.class
 )
 public class ExtractIlluminaBarcodes extends CommandLineProgram {
+    static final String USAGE_SUMMARY = "Tool to determine the barcode for each read in an Illumina lane.  ";
+    static final String USAGE_DETAILS = "<p> This tool will identify reads containing barcodes and output statistics indicating the numbers " +
+            "of reads containing barcode-matching sequences.</p> " +
+            "<p>Illumina sequences can contain at least two types of barcodes, sample and molecular (index).  Sample barcodes are used to " +
+            "demultiplex pooled samples while index barcodes are used to differentiate multiple reads of a template when carrying out paired-end sequencing. </p>" +
+            "<p>Barcodes can be provided in the form of a list (BARCODE_FILE) or a string representing the barcode.  " +
+            "The BARCODE_FILE contains multiple fields including 'barcode_sequence_1', 'barcode_sequence_2' (optional), " +
+            "'barcode_name', and 'library_name'.  In contrast, the BARCODE argument is used for runs with reads containing a single barcode (nonmultiplexed) and can be added directly " +
+            "as a string of text e.g. BARCODE=CAATAGCG.</p>" +
+            "" +
+            "<p>Data is output per lane/tile within the BaseCalls directory with the file name format of 's_{lane}_{tile}_barcode.txt'.  " +
+            "These files contain the following tab-separated columns:" +
+            "<ul> " +
+            "<li>Read subsequence at barcode position</li>" +
+            "<li>Y or N indicating if there was a barcode match</li>" +
+            "<li>Matched barcode sequence (empty if read did not match one of the barcodes)</li>  " +
+            "</ul>" +
+            "If there is no match but we're close to the threshold of calling it a match, we output the barcode that would have been " +
+            "matched but in lower case.  Threshold values can be adjusted to accommodate barcode sequence mismatches from the reads." +
+            "<ul>" +
+            "<li>Note 1: The order of specification of barcodes can cause arbitrary differences in output for poorly matching barcodes</li>" +
+            "<li>Note 2: Molecular barcodes (M in the read structure) are not the barcode being extracted here and will be ignored</li> " +
```

Comment:
This note should be moved to right after you say there's two types of barcodes on line 86.